performance_test/src/communication_abstractions/fast_rtps_communicator.hpp

```diff
@@ -161,8 +162,8 @@ class FastRTPSCommunicator : public Communicator
       wparam.topic.topicName = Topic::topic_name();
       wparam.topic.historyQos.kind = qos.history_kind();
       wparam.topic.historyQos.depth = qos.history_depth();
-      wparam.topic.resourceLimitsQos.max_samples = 100;
-      wparam.topic.resourceLimitsQos.allocated_samples = 100;
+      wparam.topic.resourceLimitsQos.max_samples = qos.history_depth();
```

Comment:
This looks counterintuitive, the name of the function does not match the name of the struct member. I'd add a comment here explaining why, or perhaps just move the comment from above.

Reply:
Added a new method to the qos adapter and used it here. I think it makes more sense now.